Documentation: Improve page about the HTTP endpoint/interface (#16733)
crate/crate: docs/interfaces/http.rst

====== =====================================================================

.. _http-bulk-ops:

Bulk operations
===============

> The REST endpoint allows to issue bulk operations which are executed as single

Reviewer: ```suggestion
The HTTP endpoint allows to issue bulk operations which are executed as single
```
Not directly related to your changes, but while we are at it: We refer to the endpoint as "HTTP endpoint" everywhere else on that page, and it is not a REST endpoint. A REST endpoint (or rather REST API) would support the different HTTP methods (GET/POST/PATCH/...) to perform specific actions on a resource identified in the URL.
 
A RESTful HTTP POST would need to look something like `HTTP POST https://cratedb-cluster.com:4200/my_schema/my_table` rather than taking arbitrary SQL on a static endpoint.

Author: Thanks. I think @mfussenegger already considered your suggestion within his suggestions, that have just been applied.

      ]
    }

.. rubric:: Error handling

> In general, every bulk operation item will be executed, independent if one of
> the other operations fails. However, there are two kinds of errors that will be
> handled differently, depending on in which of the query execution phases they
> are determined.

Reviewer 2: ```suggestion
In general, each bulk operation item will be executed independently, unless an error is detected during the analysis phase, in which case no operations will be executed. There are two kinds of errors that are handled differently depending on the query execution phase in which they are encountered:
```

      ]
    }

.. rubric:: Error handling

> There are two kinds of error behaviors for bulk requests:

Reviewer 3: The paragraph above in "Bulk errors" should also be removed. The information is otherwise duplicated

Author: You mean like this?
```suggestion
```

Currently, I don't see any redundant information with that sentence. Most probably, you are referring to something different, and I am just not seeing it?

I will be happy to make any changes, but I need further advise. Thanks!

Reviewer 3: No I mean like this:

```diff
diff --git a/docs/interfaces/http.rst b/docs/interfaces/http.rst
index 790dc53b1bb..73b464511dc 100644
--- a/docs/interfaces/http.rst
+++ b/docs/interfaces/http.rst
@@ -506,39 +506,10 @@ statements are:
 .. _http-bulk-errors:
 
 Bulk errors
 -----------
 
-If a bulk operation fails, the resulting row count will be ``-2`` and the
-resulting object may contain an ``error_message`` depending on the resulting
-error::
-
-    sh$ curl -sS -H 'Content-Type: application/json' \
-    ... -X POST '127.0.0.1:4200/_sql' -d@- <<- EOF
-    ... {
-    ...   "stmt": "INSERT into locations (name, id) values (?,?)",
-    ...   "bulk_args": [
-    ...     ["Mars", 1341],
-    ...     ["Sun", 1341]
-    ...   ]
-    ... }
-    ... EOF
-    {
-      "cols": [],
-      "duration": ...,
-      "results": [
-        {
-          "rowcount": 1
-        },
-        {
-          "rowcount": -2
-        }
-      ]
-    }
-
-.. rubric:: Error handling
-
 There are two kinds of error behaviors for bulk requests:
 
 1. **Analysis error:** Occurs if the statement is invalid, either due to syntax
    errors or semantic errors identified during the analysis phase before the
    execution starts. In this case the **whole** operation fails and you'll get
```
